tests/std/tests/P2278R4_basic_const_iterator/test.cpp

```diff
@@ -298,7 +298,89 @@ constexpr void instantiation_test() {
     instantiator::call<test_iterator<contiguous_iterator_tag, CanDifference::yes>>();
 }
 
-int main() {
-    static_assert((instantiation_test(), true));
+template <class T>
+struct tracking_input_iterator {
+public:
+    using value_type      = remove_const_t<T>;
+    using difference_type = ptrdiff_t;
+
+    constexpr tracking_input_iterator() = default;
+
+    constexpr tracking_input_iterator(const tracking_input_iterator&) : copied{true} {}
+
+    constexpr tracking_input_iterator(tracking_input_iterator&&) : moved{true} {}
+
+    template <class U>
+        requires convertible_to<const U&, T>
+    constexpr tracking_input_iterator(const tracking_input_iterator<U>&) : copied{true} {}
+
+    template <class U>
+        requires convertible_to<U, T>
+    constexpr tracking_input_iterator(tracking_input_iterator<U>&&) : moved{true} {}
+
+    tracking_input_iterator& operator=(const tracking_input_iterator&) = default;
+    tracking_input_iterator& operator=(tracking_input_iterator&&)      = default;
+
+    constexpr tracking_input_iterator& operator++(); // not defined
+    constexpr void operator++(int); // not defined
+    T& operator*() const; // not defined
+
+    constexpr bool is_moved() const {
+        return moved;
+    }
+
+    constexpr bool is_copied() const {
+        return copied;
+    }

```

Comment:
Style suggestion: This order is opposite of every other occurrence in the class.